.gitlab-ci.yml

```diff
@@ -448,6 +449,65 @@ check-polkadot-companion-status:
   script:
     - ./.maintain/gitlab/check_polkadot_companion_status.sh
 
+pr-docker-build:            &pr-docker-build
+  stage:                           build-pr
+  tags: 
+    - kubernetes-parity-build
+  variables:
+    DOCKER_HOST:                   tcp://localhost:2375
+    DOCKER_DRIVER:                 overlay2
+    PRODUCT:                       substrate
+    DOCKERFILE:                    $PRODUCT.Dockerfile
+    CONTAINER_IMAGE:               paritypr/$PRODUCT
+  environment:
+    name:                          parity-chaosnet
+  image:                           docker:stable
+  services:
+    - docker:dind
+  before_script:
+    - test "$DOCKER_CHAOS_USER" -a "$DOCKER_CHAOS_TOKEN"
+        || ( echo "no docker credentials provided"; exit 1 )
+    - docker login -u "$DOCKER_CHAOS_USER" -p "$DOCKER_CHAOS_TOKEN"
+    - docker info
+  script:
+    - cd ./artifacts/$PRODUCT/
+    - VERSION="ci-${CI_COMMIT_SHORT_SHA}"
+    - echo "${PRODUCT} version = ${VERSION}"
+    - test -z "${VERSION}" && exit 1
+    - docker build
+      --build-arg VCS_REF="${CI_COMMIT_SHA}"
+      --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')"
+      --tag $CONTAINER_IMAGE:$VERSION
+      --file $DOCKERFILE .
+    - docker push $CONTAINER_IMAGE:$VERSION
+  after_script:
+    - docker logout
+
+#### stage:                        chaos
+
+chaos-test-singlenodeheight:
+  stage:                           chaos
+  tags: 
+    - parity-chaos
+  variables:
+    PRODUCT:                       substrate
+    DOCKERFILE:                    $PRODUCT.Dockerfile
+    CONTAINER_IMAGE:               paritypr/$PRODUCT
+    KEEP_NAMESPACE:                0
+    NAMESPACE:                     "substrate-ci-${CI_COMMIT_SHORT_SHA}-${CI_PIPELINE_ID}"
```

Comment:
```suggestion
    NAMESPACE:                     "substrate-ci-${CI_COMMIT_SHORT_SHA}-${CI_PIPELINE_ID}"
  interruptible:                          true
```

Reply:
You can't just resolve stuff without discussing it.

Reply:
please add this suggestion

Reply:
This is resolved.

Reply:
Line 572 is duplicated.